pkg/clients/eks/nodegroup.go

```diff
@@ -54,6 +54,11 @@ func GenerateCreateNodeGroupInput(name string, p *v1alpha1.NodeGroupParameters)
 			MinSize:     p.ScalingConfig.MinSize,
 			MaxSize:     p.ScalingConfig.MaxSize,
 		}
+
+		// If desiredSize is not set, default to minSize
+		if p.ScalingConfig.DesiredSize == nil {
+			c.ScalingConfig.DesiredSize = p.ScalingConfig.MinSize
```

Comment:
Hmm I don't really feel comfortable having this controller make that decision but it seems we're required to find a value for API request while allowing user to leave it as `nil` to handle the autoscaler problem and in creation case, it becomes a chicken-egg problem. Could you explain that as comment? Maybe refer to the issue as well?

Reply:
Good point, should be resolved